node/node.go

```diff
@@ -346,21 +390,29 @@ func (node *AlgorandFullNode) ListeningAddress() (string, bool) {
 // Stop stops running the node. Once a node is closed, it can never start again.
 func (node *AlgorandFullNode) Stop() {
 	node.mu.Lock()
-	defer node.mu.Unlock()
+	defer func() {
+		node.mu.Unlock()
+		node.waitMonitoringRoutines()
+	}()
 
 	node.net.ClearHandlers()
-
-	node.txHandler.Stop()
 	node.net.Stop()
-	node.algorandService.Shutdown()
-	node.syncer.Stop()
-	node.txPoolSyncer.Stop()
-	node.ledgerService.Stop()
+	if node.catchpointCatchupService != nil {
+		node.catchpointCatchupService.Stop()
+	} else {
+		node.txHandler.Stop()
```

Comment:
does startup/shutdown order matter?

Reply:
Some of them might. For example, you don't want the network to send messages to the tx handler while it's being stopped.  (which is why we clear the handlers before ).
I'm sure there are other cases as well.
Generally speaking, I haven't modified the order unless there was a reason to do so.